gcloud-java-compute/src/main/java/com/google/gcloud/compute/ImageInfo.java

```diff
@@ -0,0 +1,416 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.services.compute.model.Image;
+import com.google.common.base.Function;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+import org.joda.time.format.DateTimeFormatter;
+import org.joda.time.format.ISODateTimeFormat;
+
+import java.io.Serializable;
+import java.math.BigInteger;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A Google Compute Engine Image. An image contains a boot loader, an operating system and a root
+ * file system that is necessary for starting an instance. Compute Engine offers publicly-available
+ * images of certain operating systems that you can use, or you can create a custom image. A custom
+ * image is an image created from one of your virtual machine instances that contains your specific
+ * instance configurations. Use {@link DiskImageConfiguration} to create an image from an existing
+ * disk. Use {@link RawImageConfiguration} to create an image from a raw image stored in Google
+ * Cloud Storage.
+ *
+ * @see <a href="https://cloud.google.com/compute/docs/images">Images</a>
+ */
+public class ImageInfo implements Serializable {
+
+  static final Function<Image, ImageInfo> FROM_PB_FUNCTION =
+      new Function<Image, ImageInfo>() {
+        @Override
+        public ImageInfo apply(Image pb) {
+          return ImageInfo.fromPb(pb);
+        }
+      };
+  static final Function<ImageInfo, Image> TO_PB_FUNCTION =
+      new Function<ImageInfo, Image>() {
+        @Override
+        public Image apply(ImageInfo image) {
+          return image.toPb();
+        }
+      };
+
+  private static final long serialVersionUID = -1061916352807358977L;
+  private static final DateTimeFormatter TIMESTAMP_FORMATTER = ISODateTimeFormat.dateTime();
+
+  private final String id;
+  private final ImageId imageId;
+  private final Long creationTimestamp;
+  private final String description;
+  private final ImageConfiguration configuration;
+  private final Status status;
+  private final Long diskSizeGb;
```

Comment:
Is [archiveSizeBytes](https://cloud.google.com/compute/docs/reference/latest/images#archiveSizeBytes) something we care about, or was it deliberately left out?